sympy/matrices/matrices.py

```diff
@@ -318,13 +269,15 @@ def elementary_row_op(self, op="n->kn", row=None, k=None, row1=None, row2=None):
 
         op, row, k, row1, row2 = self._normalize_op_args(op, row, k, row1, row2, "row")
 
-        # now that we've validated, we're all good to dispatch
-        if op == "n->kn":
-            return self._eval_row_op_multiply_row_by_const(row, k)
-        if op == "n<->m":
-            return self._eval_row_op_swap(row1, row2)
-        if op == "n->n+km":
-            return self._eval_row_op_add_multiple_to_other_row(row, k, row2)
+        entry = {
+            "n->kn": lambda i, j: k * self[i, j] if i == row else self[i, j],
+            "n<->m": lambda i, j: self[row2, j] if i == row1 else self[row1, j] \
+                    if i == row2 else self[i, j],
+            "n->n+km": lambda i, j: self[i, j] + k * self[row2, j] if i == row \
+                    else self [i, j]
+        }
+
+        return self._new(self.rows, self.cols, entry[op])
```

Comment:
Where is the error checking for a valid `op`?

Reply:
In `_normalize_op_args`.